src/server/scripts/Northrend/zone_howling_fjord.cpp

```diff
@@ -659,6 +660,111 @@ class npc_riven_widow_cocoon : public CreatureScript
     }
 };
 
+enum NecroOverlordMezhenEvents
+{
+    EVENT_CAST_IDLE_BEAM = 1,
+    EVENT_CAST_DRAIN_SOUL,
+    EVENT_CAST_RAISE_DEAD
+};
+
+enum NecroOverlordMezhenSpells
+{
+    SPELL_IDLE_PURPLE_BEAM = 43151,
+    SPELL_DRAIN_SOUL = 32862,
+    SPELL_RAISE_DEAD = 43559
+};
+
+class npc_necro_overlord_mezhen : public CreatureScript
+{
+public:
+    npc_necro_overlord_mezhen() : CreatureScript("npc_necro_overlord_mezhen") { }
+
+    struct npc_necro_overlord_mezhenAI : public ScriptedAI
+    {
+        npc_necro_overlord_mezhenAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void Reset() override
+        {
+            events.Reset();
+
+            events.ScheduleEvent(EVENT_CAST_IDLE_BEAM, 333);
+        }
+
+        void EnterCombat(Unit* /*victim*/) override
+        {
+            events.ScheduleEvent(EVENT_CAST_DRAIN_SOUL, randtime(Seconds(0), Seconds(5)));
+            events.ScheduleEvent(EVENT_CAST_RAISE_DEAD, randtime(Seconds(0), Seconds(10)));
```

Comment:
you can remove those `randtime`, like this: `events.ScheduleEvent(EVENT_CAST_RAISE_DEAD, Seconds(0), Seconds(10));`